src/localStorage.js

```diff
@@ -0,0 +1,155 @@
+const list = JSON.parse(localStorage.getItem('list')) || [];
+const insert = document.getElementById('insert');
+const listItems = document.querySelector('.list-items');
+function ToDo(descreption, completed, index) {
+  this.descreption = descreption;
+  this.completed = completed;
+  this.index = index;
+}
+
+function removeChildNodes(listItems) {
+  while (listItems.firstChild) {
+    listItems.removeChild(listItems.firstChild);
+  }
+}
+
+function refreshStorage() {
+  localStorage.setItem('list', JSON.stringify(list));
+}
+
+function update() {
+  for (let i = 0; i < list.length; i += 1) {
+    list[i].index = i;
+    refreshStorage();
+  }
+}
+
+function clearAll() {
+  const clearAllBtn = document.querySelector('.remove-button');
+  if (listItems.childNodes.length > 0) {
+    clearAllBtn.style.display = 'block';
+  } else {
+    clearAllBtn.style.display = 'none';
+  }
+  clearAllBtn.addEventListener('click', () => {
+    for (let i = 0; i < list.length; i += 1) {
+      if (list[i].completed) {
+        list.splice(i, 1);
+        listItems.removeChild(listItems.childNodes[i]);
+        window.location.reload();
+        refreshStorage();
+        update();
+      }
+    }
+  });
+}
+
+function addToDo() {
+  const newAdd = new ToDo();
+  newAdd.descreption = document.getElementById('insert').value;
+  newAdd.completed = false;
+  list.push(newAdd);
+  update();
+}
+
+function pushContent() {
+  removeChildNodes(listItems);
+  insert.value = '';
+  insert.focus();
+  list.forEach((element) => {
+    const listItem = document.createElement('li');
+    const check = document.createElement('div');
+    check.className = 'check';
+    const input = document.createElement('input');
+    input.type = 'checkbox';
+    input.className = 'checkbox';
+    const descreForm = document.createElement('form');
+    descreForm.className = 'desc-form';
+    const descreption = document.createElement('input');
+    descreption.type = 'text';
+    descreption.className = 'descreption';
```

Comment:
- [ ] I see that you are adding an actual `function`  to add tasks, but you are on the list structure you should have just the object of an array.

![pm](https://user-images.githubusercontent.com/61562626/135652260-e7f30c6a-2cca-4ef1-81eb-9a6f542627e5.jpeg)

Reply:
> 

sorry I couldn't understand that point  It is already an object of an array

Reply:
- [ ] I think that what my colleague was trying to say is that at this point of the project you should not have a function for adding tasks 😮‍, you should have an Array with tasks and populate the HTML file with the data inside that Array.
    - Kindly take a look at the project requirements and fix this issue 🤓 
    - As you can see in this image your project is not displaying any tasks. 😿 
    ![image](https://user-images.githubusercontent.com/60789317/135669262-902d6c6b-a823-4494-b916-90e1594cfd16.png)

Reply:
> > 
> 
> sorry I couldn't understand that point  It is already an object of an array
> 

The point of it is to have like a list inside, instead of push a new one.